app/models/concerns/withdrawable.rb

```diff
@@ -0,0 +1,45 @@
+module Withdrawable
+  def page_title
+    withdrawn? ? "[Withdrawn] #{title}" : title
+  end
+
+  def withdrawn?
+    withdrawal_notice.present?
+  end
+
+  def withdrawal_notice_component
+    if withdrawn?
+      {
+        title: withdrawal_notice_title,
+        description_govspeak: withdrawal_notice["explanation"]&.html_safe,
+        time: withdrawal_notice_time,
+        lang: I18n.locale.to_s == "en" ? false : "en",
+      }
+    end
+  end
+
+private
+
+  def withdrawal_notice
+    content_store_hash["withdrawn_notice"]
+  end
+
+  def withdrawal_notice_title
+    "This #{withdrawal_notice_context.downcase} was withdrawn on #{withdrawal_notice_time}".html_safe
+  end
+
+  def withdrawal_notice_context
+    I18n.t("formats.#{schema_name}", count: 1, locale: :en)
+  end
+
+  def withdrawal_notice_time
+    ActionController::Base.helpers.tag.time(
+      english_display_date(withdrawal_notice["withdrawn_at"]),
+      datetime: withdrawal_notice["withdrawn_at"],
+    )
+  end
+
+  def english_display_date(timestamp, format = "%-d %B %Y")
+    I18n.l(Time.zone.parse(timestamp), format:, locale: :en) if timestamp
+  end
```

Comment:
A bit suspicious of these presentational items being in a model concern, might be better as helpers?

Reply:
Travel Advice and Guides need to know the withdrawn state too, but I've tackled it a bit differently by splitting out the model and presentational concerns:

model concern: https://github.com/alphagov/frontend/pull/4225/commits/0dfd81189601511ab3d468001b35098ffec32cad

generic presenter: https://github.com/alphagov/frontend/pull/4225/commits/0590ed72bf4543ff55feae6b1539f817fb54b3b5

travel advice presenter: https://github.com/alphagov/frontend/pull/4225/commits/53ee7c310524d77dc7c83fe86ed10d5d23b2cf8d

I'm not completely happy with the approach I've taken either, so I think it would be good to have a discussion about this.